app/src/main/java/com/github/dedis/student20_pop/utility/qrcode/QRCodeListener.java

```diff
@@ -1,9 +1,11 @@
 package com.github.dedis.student20_pop.utility.qrcode;
 
+import com.github.dedis.student20_pop.ui.QRCodeScanningFragment.QRCodeScanningType;
+
 /**
  * A listener that will receive detected url of a QR code
  */
 public interface QRCodeListener {
 
-    void onQRCodeDetected(String url);
+    void onQRCodeDetected(String url, QRCodeScanningType qrCodeScanningType);
 }
```

Comment:
We might want to change this in the future the give back a more general data than an url.

Reply:
I can change that to make it more generic